### DISCUSSION TITLE
Is there a way to update the colour of a accordion panel without changing anything else in it?

### DISCUSSION THREAD

**Discussion by udurraniAtPresage** [rstudio/bslib#988] [Q&A] [answered]
The documentation on accordions says:    

`.. Named arguments become attributes on the ⁠<div class="accordion">⁠ element. Unnamed arguments should be accordion_panel()s.`

I tried the following by providing the `style` argument, but it did not change the colour, instead it replaced the button with the literal style value.  

```
library(shiny)
library(bslib)

ui <- fluidPage(
  theme = bs_theme(version = 5),
  accordion(
    id = "accordion",
    accordion_panel(
      "Panel 1",
      actionButton("saveBtn1", "Save")
    ),
    accordion_panel(
      "Panel 2",
      actionButton("saveBtn2", "Save")
    )
  )
)

server <- function(input, output, session) {
  observeEvent(input$saveBtn1, {
    accordion_panel_update("accordion", "Panel 1", style = "background-color: green;")
  })

  observeEvent(input$saveBtn2, {
    accordion_panel_update("accordion", "Panel 2", style = "background-color: green;")
  })
}

shinyApp(ui, server)

```

**Comment by gadenbuie** [comment, marked as answer]
Hi @udurraniAtPresage, thanks for the question! Unfortunately, `accordion_panel_set()` can currently only update the contents of a panel and not attributes. We'll certainly consider adding that feature in the future, but for now I've updated the documentation of `accordion_panel_set()` in #989 to reflect that `...` are the unnamed child elements that will become the panel content.

**Reply by udurraniAtPresage** [reply]
Thanks. Do you have any suggestions for updating the panel colour? I guess I need to use a javascript function to update css. But please share if you have any other ideas.

**Reply by gadenbuie** [reply]
Here's another idea: instead of updating the panel color, you can add an icon to the completed panels. Here's an app with that behavior ([try it on shinylive](https://shinylive.io/r/editor/#code=NobwRAdghgtgpmAXGKAHVA6ASmANGAYwHsIAXOMpMAGwEsAjAJykYE8AKAZwAtaJWAlAB0IdJiw71OY4RBEBXWgAIAPAFolAM2qKAJgAUoAczjsRSpaW5x4SgLxKpAfSs3TANziNOtEvaUArAK45kpQBMSMur4QZhAWFrS6-kIoEURRMakh8QnhkdEkTqhQEHDUcQkJqYZl1EoAjNmhVeGkMQBC8qSkJGZgnFCeHaQQTXhKqQDKQ3CpslXBLWHpmUUldZVVk2C15UoATM25rQTtJF09famDw6NHE9Oz88sLSrKyIpxenoyqGpp5BAzjF2HxUN1cEoiN0IaQod9OD4SAIlCBQrRiBAnJx5BE4Ej-o4fFjOIhEM5MddCNYCABrNQEWiMAjUOBqTS0ajUbJKVlQJF2VLkAAepDUuPxSN57VIbJSYAAwkQYKg2eRdC8TpNckQpD84ABRTxkMEQOEAElucBGYyh6O1+QyhWxG3KTnkqF0UHI-Sdazkj12pX24yhVPiDgjOLxBAJnDeFn9LuKIeoOLgpD9qxdvJqacOWosAF9Prr9YxPMaKFnwd0rbNbQd7ctkzFU3UPV6faZUm2SHng3VCxMI-5o5K40i3qWRMWRF9ePwAILodiKBEGxiyMDFgC6QA)), and it also moves from Panel 1 to Panel 2 when you complete the first panel:

```r
library(shiny)
library(bslib)

ui <- fluidPage(
  theme = bs_theme(version = 5),
  accordion(
    id = "accordion",
    accordion_panel(
      "Panel 1",
      actionButton("saveBtn1", "Save")
    ),
    accordion_panel(
      "Panel 2",
      actionButton("saveBtn2", "Save")
    )
  )
)

server <- function(input, output, session) {
  icon_success <- bsicons::bs_icon("check-circle-fill", class="text-success", title = "Completed")
    
  observeEvent(input$saveBtn1, {
    accordion_panel_update("accordion", "Panel 1", icon = icon_success)
    accordion_panel_set("accordion", "Panel 2")
  })

  observeEvent(input$saveBtn2, {
    accordion_panel_update("accordion", "Panel 2", icon = icon_success)
  })
}

shinyApp(ui, server)
```

**Reply by udurraniAtPresage** [reply]
This is neat! Thank you.
